leetcodePython/SlidingWindow/slidingwindow_567.py

```diff
@@ -0,0 +1,33 @@
+from collections import Counter
+
+
+class Solution:
+    def checkInclusion(self, s1: str, s2: str) -> bool:
+        if len(s1) > len(s2):
+            return False
+        s1_freq = Counter(s1)
+        window_freq = Counter()
+        left = 0
+        matched = 0
+        for right in range(len(s2)):
+            right_ch = s2[right]
+            if right_ch in s1_freq:
+                window_freq[right_ch] += 1
+                if window_freq[right_ch] == s1_freq[right_ch]:
+                    matched += 1
+            while right - left + 1 >= len(s1):
```

Comment:
Every time we have to shrink the window size to s1 size. Because we want the window to be permutation of s1